src/config.rs

```diff
@@ -77,16 +77,24 @@ pub struct FnmConfig {
     corepack_enabled: bool,
 
     /// Resolve `engines.node` field in `package.json` whenever a `.node-version` or `.nvmrc` file is not present.
-    /// Experimental: This feature is subject to change.
+    /// This feature is enabled by default. To disable it, provide `--resolve-engines=false`.
+    ///
     /// Note: `engines.node` can be any semver range, with the latest satisfying version being resolved.
+    /// Note 2: If you disable it, please open an issue on GitHub describing _why_ you disabled it.
+    ///         In the future, disabling it might be a no-op, so it's worth knowing any reason to
+    ///         do that.
     #[clap(
         long,
         env = "FNM_RESOLVE_ENGINES",
         global = true,
         hide_env_values = true,
         verbatim_doc_comment
     )]
-    resolve_engines: bool,
+    #[expect(
+        clippy::option_option,
+        reason = "clap Option<Option<T>> supports --x and --x=value syntaxes"
+    )]
+    resolve_engines: Option<Option<bool>>,
```

Comment:
Why is it `Option<Option<bool>>` and not `Option<bool>`?

Reply:
because `Option<Option<bool>>` means that `--resolve-engines` will work and so will `--resolve-engines=false`

Reply:
Ugh

So `--resolve-engines` is `Some(None)`, and `--resolve-engines=false` is `Some(Some(false))`?